libraries/bsg_manycore_config.cpp

```diff
@@ -135,6 +135,11 @@ int hb_mc_config_init(const hb_mc_config_raw_t raw[HB_MC_CONFIG_MAX],
         idx = raw[HB_MC_CONFIG_DEVICE_HOST_INTF_COORD_Y];
         config->host_interface.y = idx;
 
+        idx = raw[HB_MC_CONFIG_ORIGIN_COORD_X];
+        config->origin.x = idx;
+
+        idx = raw[HB_MC_CONFIG_ORIGIN_COORD_Y];
+        config->origin.y = idx;
```

Comment:
I would run CHECK_FIELD after each of these. They should probably be >0 and a power of two.